src/pages/Environments.js

```diff
@@ -254,27 +254,29 @@ const Environments = () => {
     return getWorkspaceCell(saturnWorkspaceNamespace, saturnWorkspaceName, null, shouldWarn)
   }
 
-  const getDetailsPopup = (cloudEnvName, billingId, disk, creator) => {
+  const getDetailsPopup = (cloudEnvName, billingId, disk, creator, workspaceId) => {
     return h(PopupTrigger, {
       content: div({ style: { padding: '0.5rem' } }, [
         div([strong(['Name: ']), cloudEnvName]),
-        div([strong(['Billing Id: ']), billingId]),
+        div([strong(['Billing ID: ']), billingId]),
+        !!workspaceId && div([strong(['Workspace ID: ']), workspaceId]),
         !shouldFilterRuntimesByCreator && div([strong(['Creator: ']), creator]),
         !!disk && div([strong(['Persistent Disk: ']), disk.name])
       ])
     }, [h(Link, ['view'])])
   }
 
   const renderDetailsApp = (app, disks) => {
-    const { appName, diskName, googleProject } = app
+    const { appName, diskName, googleProject, auditInfo: { creator } } = app
     const disk = _.find({ name: diskName }, disks)
-    return getDetailsPopup(appName, googleProject, disk)
+    // There's no current plans for workspaceId to be defined for apps
+    return getDetailsPopup(appName, googleProject, disk, creator, undefined)
```

Comment:
Hmm... so currently we rely on leo to attach the workspaceId to the runtime, which is only done for azure VMs since we get this info through WSM. It should be possible to use the labels on the app/runtime (workspaceNamespace and workspaceName) to retrieve this ID from rawls UI-side though. Let me investigate...

Reply:
I'm referring to the UUID for workspace, is that what you're thinking as well?

Reply:
Yes. This information is optional from the leo response (and only populated for azure runtime currently). I now get this info from rawls directly instead of omitting it if leo hasn't returned it. Check it out now. It has a bit longer of a load time, but it seems worth it.